### DISCUSSION TITLE
5G SA Network

### DISCUSSION THREAD

**Discussion by sambak22** [srsran/srsRAN_Project#538] [General Help]
I have successfully built srsRAN 5G project in Ubuntu 22.04. Further, I have also connected USRP B205 Mini. When I am running ./gnB compiler using B200 TDD config file, I am unable to get following output:
Available radio types: uhd.

--== srsRAN gNB (commit 77be7d339) ==--

[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.2.0.HEAD-0-g197cdc4f
Making USRP object with args 'type=b200'
Cell pci=1, bw=10 MHz, dl_arfcn=632628 (n78), dl_freq=3489.42 MHz, dl_ssb_arfcn=632640, ul_freq=3489.42 MHz

==== gNodeB started ===
Type <t> to view trace

Please guide me to get through this step. Also, please assist me in setting up 5G SA. I have successfully built open5gs as core network and tested it using UERANSIM. I would want to run open5gs and srsRAN in same machine installed with Ubuntu 22.04.

Thank you

**Comment by pgawlowicz** [comment]
Please check our tutorials that show how to deploy a 5G network: https://docs.srsran.com/projects/project/en/latest/tutorials/source/index.html

**Reply by sambak22** [reply]
Thank you for your prompt reply. However, I wish to highlight that I am stuck at running gnb as mentioned above. I am unable to get the desired output. I have followed all the steps till connecting USRP B205 Mini and then executing ./gnb -c <config file>. After executing the ./gnb code, I get following output:
The PRACH detector will not meet the performance requirements with the configuration {Format B4, ZCZ 0, SCS 30kHz, Rx ports 1}. Lower PHY in dual executor mode. --== srsRAN gNB (commit 2f90c8b60) ==-- Connecting to AMF on 127.0.1.100:38412 Failed to connect SCTP socket to 127.0.1.100:38412. error="Connection refused" timeout=0ms sr GNB ERROR: Failed to create SCTP gateway.

Please assist me to resolve the same.

Thank you, in anticipation,
Regards

**Reply by pgawlowicz** [reply]
do you run a core network (eg. Open5gs) at 127.0.1.100?

**Reply by sambak22** [reply]
How do I check that? Please guide me. Also, after connecting USRP B205 Mini, the desired output must be as under:

Available radio types: uhd.

--== srsRAN gNB (commit https://github.com/srsran/srsRAN_Project/commit/77be7d3391d45fb606aec4f0c16907390b49b0b0) ==--

[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.2.0.HEAD-0-g197cdc4f
Making USRP object with args 'type=b200'
Cell pci=1, bw=10 MHz, dl_arfcn=632628 (n78), dl_freq=3489.42 MHz, dl_ssb_arfcn=632640, ul_freq=3489.42 MHz

However, I am not getting the same. So, I am unable to resolve this. I have multiple times installed the UHD drivers and srsRAN 5G.

Alternatively, I was looking for option of installing srsRAN 4G. Will it meet my purpose of deploying 5G SA network.

Regards

**Reply by pgawlowicz** [reply]
could you share your open5gs config?

btw. the best way to start working with srsRAN_Project is to follow our tutorials: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html

**Reply by sambak22** [reply]
Sir, I will share the open5gs config shortly. In the mean time, if you can confirm if I can install srsRAN 4G to deploy open5gs as 5G SA network.

**Reply by pgawlowicz** [reply]
yes, you can setup a 5G SA network with srsRAN-4G project. Please follow this tutorial:
https://docs.srsran.com/projects/4g/en/latest/app_notes/source/5g_sa_amari/source/index.html

However, it is recommended to use the new srsRAN_Project to setup a 5G SA network.

**Reply by sambak22** [reply]
[amf.conf.txt](https://github.com/srsran/srsRAN_Project/files/14840151/amf.conf.txt)
Sir,
You are kindly requested to review the amf config file and revert as per your convenient time.

Thank you & Regards

**Reply by pgawlowicz** [reply]
It seems that amf is running at 127.0.0.5, so you need to update the srsgnb config accordingly.

**Reply by sambak22** [reply]
Hello Sir! I updated the srsgnb config with 127.0.0.5 but still no success.
[gnb_rf_b210_fdd_srsUE.yml.txt](https://github.com/srsran/srsRAN_Project/files/14921437/gnb_rf_b210_fdd_srsUE.yml.txt)
Also, I am getting following error message:
Connecting to AMF on 127.0.0.5:38412
Failed to connect SCTP socket to 127.0.0.5:38412. error="Connection refused" timeout=0ms
srsGNB ERROR: Failed to create SCTP gateway.

**Reply by pgawlowicz** [reply]
could you use the dockerized version of Open5GS as described here: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html ?

**Reply by pgawlowicz** [reply]
@sambak22 any update on this issue?

**Reply by sambak22** [reply]
Hello! I have followed the tutorial thoroughly but I am unable to resolve the above mentioned issue. Please help me. Thank You. Please find attached config files of gnb, anf and upf for your reference.
[gnb.yaml.txt](https://github.com/srsran/srsRAN_Project/files/15204670/gnb.yaml.txt)
[amf.yaml.txt](https://github.com/srsran/srsRAN_Project/files/15204701/amf.yaml.txt)
[upf.yaml.txt](https://github.com/srsran/srsRAN_Project/files/15204707/upf.yaml.txt)
